test/fakeintake/fixtures/process.go

```diff
@@ -13,6 +13,20 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func encode(t *testing.T, typ agentmodel.MessageType, body agentmodel.MessageBody) []byte {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
what I usually do is get a real dump from a test environment with fakeintake. You can get raw data from any endpoint on the fakeintake and create fixture binary files out of them. Could automate that with a script maybe.

Reply:
That's a good idea—I had feedback on an earlier PR that just adding binary files to a PR is a bit too obfuscated, but creating a script to reproduce it would definitely work. I'll try to add it in a follow up

Reply:
I have an idea for this: what about having a table file, like a csv, with on one column the readable, json dump version and on the other side the binary version ? We could generate such files. I am watching https://www.youtube.com/watch?v=8hQG7QlcLBk that talks about golden files, this could be a candidate